src/implementations/config/create.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  Issuer,
+  RevocationType,
+  IdentityProofType,
+  OpenAttestationDocument,
+} from "@govtechsg/open-attestation/dist/types/__generated__/schema.2.0";
+import fs from "fs";
+import path from "path";
+import { info, success } from "signale";
+import { deployDocumentStore } from "../../implementations/deploy/document-store";
+import { deployTokenRegistry } from "../../implementations/deploy/token-registry";
+import { readFile } from "../../implementations/utils/disk";
+import { create as createWallet } from "../../implementations/wallet/create";
+import tradetrustConfig from "./templates/tradetrust.json";
+import { highlight } from "../../utils";
+import { handler as createTemporaryDns } from "../../commands/dns/txt-record/create";
+import { CreateConfigCommand } from "../../commands/config/config.type";
+
+interface ConfigFile {
+  wallet: EncryptedWallet | AwsKmsWallet;
+  forms: Form[];
+}
+
+interface EncryptedWallet {
+  type: "ENCRYPTED_JSON";
+  encryptedJson: string;
+}
+
+interface AwsKmsWallet {
+  type: "AWS_KMS";
+  accessKeyId: string;
+  region: string;
+  kmsKeyId: string;
+}
+
+interface Form {
+  type: "VERIFIABLE_DOCUMENT" | "TRANSFERABLE_RECORD";
+  defaults: OpenAttestationDocument;
+}
+
+interface TypesOfForms {
+  type: "VERIFIABLE_DOCUMENT" | "TRANSFERABLE_RECORD";
+  identityProofTypes: (IdentityProofType | undefined)[];
+}
+
+const sandboxEndpointUrl = "https://sandbox.fyntech.io";
+
+export const create = async ({
+  encryptedWalletPath,
+  outputDir,
+  configType,
+  configTemplatePath,
+}: CreateConfigCommand): Promise<string> => {
+  let walletPath = "";
+  if (!encryptedWalletPath) {
+    info(`Wallet file not provided, please enter password to create a new wallet`);
+    const createWalletParams = {
+      outputFile: path.join(outputDir, "wallet.json"),
+      fund: "ropsten",
+    };
+    walletPath = await createWallet(createWalletParams);
+    info(`Wallet created at ${walletPath}`);
+  }
+  const walletFilePath = walletPath || encryptedWalletPath;
+  const wallet = await readFile(walletFilePath);
+  const walletObject = JSON.parse(wallet);
+
+  info(`Wallet detected at ${walletFilePath}`);
+
+  const configFile: ConfigFile = await selectTemplateFile(configType, configTemplatePath);
+  if (configFile.wallet.type === "ENCRYPTED_JSON") {
+    configFile.wallet.encryptedJson = wallet;
+  }
+
+  const formsInTemplate = configFile.forms;
+
+  const createDocumentStore = async (): Promise<string> => {
+    info(`Enter password to continue deployment of Document Store`);
+    const deployDocumentStoreParams = {
+      encryptedWalletPath: walletFilePath,
+      network: "ropsten",
+      gasPriceScale: 1,
+      dryRun: false,
+      storeName: "Document Store",
+    };
+    const { contractAddress } = await deployDocumentStore(deployDocumentStoreParams);
+    success(`Document store deployed, address: ${highlight(contractAddress)}`);
+    return contractAddress;
+  };
```

Comment:
can move these functions out and maybe break them down to what they are?

createDocumentStore will return created documentStore, then if need be get contract address from that. currently the function reads as createDocumentStore but it returns its contract address

Reply:
ok